src/strat_engine/engine.rs

```diff
@@ -45,9 +47,30 @@ impl Engine for StratEngine {
             return Err(EngineError::Stratis(ErrorEnum::AlreadyExists(name.into())));
         }
 
-        let pool_uuid = Uuid::new_v4();
-        let bds = try!(BlockDev::initialize(&pool_uuid, blockdev_paths, MIN_MDA_SIZE, true));
-        let pool = StratPool::new(name, pool_uuid, &bds, raid_level);
+        let mut devices = BTreeSet::new();
+        for path in blockdev_paths {
+            let dev = try!(Device::from_str(&path.to_string_lossy()));
+            devices.insert(dev);
+        }
+
+        if devices.len() != blockdev_paths.len() {
```

Comment:
If I write out a grocery list that says:

* milk
* peanuts
* prunes
* pretzels
* prunes

I would rather that the person with the list got me what I'ld asked for than that they returned empty handed because I had listed prunes twice.